crates/board/src/usb/ctap.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! CTAP HID interface.
+
+use usb_device::bus::UsbBus;
+use usbd_ctaphid::CtapHid;
+
+use super::serial::{self, HasSerial};
+
+/// CTAP event.
+#[derive(Debug, PartialEq, Eq)]
+pub enum Event {
+    /// There might be data to read.
+    ///
+    /// This is only guaranteed to be triggered after a short read.
+    Read,
+
+    /// It might be possible to write data.
+    ///
+    /// This is only guaranteed to be triggered after a short write.
+    Write,
+}
+
+impl<B: crate::Api> From<Event> for crate::Event<B> {
+    fn from(event: Event) -> Self {
+        super::Event::Ctap(event).into()
+    }
+}
+
+pub trait Api: Send {
```

Comment:
We should add the event support with `enable()` and `disable()`. This is needed by OpenSK.